rust/kernel/module_param.rs

```diff
@@ -455,13 +467,14 @@ impl ModuleParam for StringParam {
     fn try_from_param_arg(arg: Option<&'static [u8]>) -> Option<Self> {
         // SAFETY: It is always safe to call [`slab_is_available`](../../../include/linux/slab.h).
         let slab_available = unsafe { crate::bindings::slab_is_available() };
-        arg.map(|arg| {
+        arg.and_then(|arg| {
             if slab_available {
                 let mut vec = alloc::vec::Vec::new();
+                vec.try_reserve(arg.len()).ok()?;
```

Comment:
Should this be `try_reserve_exact()`? i.e. do we expect further allocations?

Reply:
Ah, yep. `exact` is better.